lib/Language/Souffle/Interpreted.hs

```diff
@@ -91,17 +93,17 @@ instance MonadSouffle SouffleM where
       let outDir  = souffleTempDir </> "out"
       createDirectoryIfMissing True factDir
       createDirectoryIfMissing True outDir
-      -- TODO: Use system commands to locate the interpreter
-      -- TODO: souffle not found => Nothing
-      souffleBin <- fromMaybe "souffle" <$> lookupEnv "SOUFFLE_BIN"
-      fmap (Just . Handle) $ newIORef $ HandleData
-        { soufflePath = souffleBin
-        , basePath    = souffleTempDir
-        , factPath    = factDir
-        , outputPath  = outDir
-        , datalogExec = datalogExecutable
-        , noOfThreads = 1
-        }
+      mSouffleBin <- mplus <$> lookupEnv "SOUFFLE_BIN"
+                           <*> locateSouffle
+      forM mSouffleBin $ \souffleBin -> do
```

Comment:
I never considered `forM` for `Maybe`, that is a neat trick for avoiding a case expression!